tiles/src/main/java/com/protomaps/basemap/layers/Pois.java

```diff
@@ -3,85 +3,308 @@
+          kind = sf.getString("leisure");
+        } else if (sf.hasTag("railway")) {
+          kind = sf.getString("railway");
+        } else if (sf.hasTag("shop")) {
+          kind = sf.getString("shop");
+        } else if (sf.hasTag("tourism")) {
+          kind = sf.getString("tourism");
+        // Boundary is most generic, so place last else we loose out
+        // on nature_reserve detail versus all the protected_area
+        } else if (sf.hasTag("boundary")) {
+          kind = sf.getString("boundary");
+        }
+      }
+
+      if( sf.hasTag("boundary", "national_park") &&
+              !(sf.hasTag("operator", "United States Forest Service", "US Forest Service", "U.S. Forest Service", "USDA Forest Service", "United States Department of Agriculture", "US National Forest Service", "United State Forest Service", "U.S. National Forest Service") ||
+                      sf.hasTag("protection_title", "Conservation Area", "Conservation Park", "Environmental use", "Forest Reserve", "National Forest", "National Wildlife Refuge", "Nature Refuge", "Nature Reserve", "Protected Site", "Provincial Park", "Public Access Land", "Regional Reserve", "Resources Reserve", "State Forest", "State Game Land", "State Park", "Watershed Recreation Unit", "Wild Forest", "Wilderness Area", "Wilderness Study Area", "Wildlife Management", "Wildlife Management Area", "Wildlife Sanctuary")
+              ) &&
+              ( sf.hasTag("protect_class", "2", "3") ||
+                      sf.hasTag("operator", "United States National Park Service", "National Park Service", "US National Park Service", "U.S. National Park Service", "US National Park service") ||
+                      sf.hasTag("operator:en", "Parks Canada") ||
+                      sf.hasTag("designation", "national_park") ||
+                      sf.hasTag("protection_title", "National Park")
+              )
+      ) {
+        kind = "national_park";
+        min_zoom = 11;
+      }
+
+      if( sf.hasTag("cuisine") ) {
+        kind_detail = sf.getString("cuisine");
+      } else if ( sf.hasTag("religion") ) {
+        kind_detail = sf.getString("religion");
+      } else if ( sf.hasTag("sport") ) {
+        kind_detail = sf.getString("sport");
       }
 
       // try first for polygon -> point representations
-      if (sf.canBePolygon()) {
+      if (sf.canBePolygon() && sf.hasTag("name") && sf.getString("name") != null) {
+        Double way_area = 0.0;
+        try {
+          way_area = sf.worldGeometry().getEnvelopeInternal().getArea() / WORLD_AREA_FOR_70K_SQUARE_METERS;
+        } catch(GeometryException e) {
+          System.out.println(e);
+        }
+
+        Double height = 0.0;
+        try { height = sf.getString("height") == null ? 0.0 : parseDoubleOrNull(sf.getString("height")); } catch(Exception e) {
+          System.out.println("Problem getting height");
+        }
+
+        // Area zoom grading overrides the kind zoom grading in the section above.
```

Comment:
If the building or landuse polygon was large, then show the POI earlier than default for nodes...